src/serialization.cpp

```diff
@@ -240,8 +347,24 @@ void compress(const SharedBuffer<u8> &data, std::ostream &os, u8 version)
 	os.write((char*)&current_byte, 1);
 }
 
+void compress(const SharedBuffer<u8> &data, std::ostream &os, u8 version, int level)
```

Comment:
this function must clearly have the compression type in mind as we have 2 types of compression here

Reply:
It does that automatically depending on version (we have an old, third compression type already)